File: backend/app/schemas/sync_job.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class PriceSyncRequest(BaseModel):
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    scope: str = "all"
    products: Optional[List[List[str]]] = None


class SyncJobResponse(BaseModel):
    id: int
    job_type: str
    status: str
    params: Optional[dict] = None
    total: int = 0
    done: int = 0
    success_count: int = 0
    failed_count: int = 0
    skipped_count: int = 0
    error_message: Optional[str] = None
    triggered_by: str = "manual"
    created_at: Optional[datetime] = None
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class NavSyncDetailResponse(BaseModel):
    id: int
    job_id: Optional[int] = None
    task_log_id: Optional[int] = None
    product_code: str
    market: str
    nav_date: str
    status: str
    synced_count: int = 0
    source: Optional[str] = None
    error_message: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/syncJob.ts
// 后台同步任务（对齐后端 SyncJobResponse，GET /api/sync-jobs/{id}）
export interface SyncJob {
  id: number;
  job_type: string;
  /** pending | running | success | failed */
  status: string;
  params?: Record<string, unknown> | null;
  total: number;
  done: number;
  success_count: number;
  failed_count: number;
  skipped_count: number;
  error_message?: string | null;
  triggered_by: string;
  created_at?: string | null;
  started_at?: string | null;
  finished_at?: string | null;
}
